pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -136,11 +147,45 @@ func (c *Controller) finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 		NodeClaimTerminationDuration.With(map[string]string{
 			metrics.NodePoolLabel: nodeClaim.Labels[v1beta1.NodePoolLabelKey],
 		}).Observe(time.Since(stored.DeletionTimestamp.Time).Seconds())
-		logging.FromContext(ctx).Infof("deleted nodeclaim")
 	}
 	return reconcile.Result{}, nil
 }
 
+func (c *Controller) ensureTerminationGracePeriodTerminationTimeAnnotation(ctx context.Context, node *v1.Node, nodeClaim *v1beta1.NodeClaim) error {
+	// if the expiration annotation is already set, we don't need to do anything
+	if _, exists := node.ObjectMeta.Annotations[v1beta1.NodeTerminationTimestampAnnotationKey]; exists {
+		return nil
+	}
+
+	nodePoolName := types.NamespacedName{Name: nodeClaim.Labels[v1beta1.NodePoolLabelKey]}
+	nodePool := &v1beta1.NodePool{}
+	if err := c.kubeClient.Get(ctx, nodePoolName, nodePool); err != nil {
```

Comment:
We should actually ignore not found errors here, ran into this while testing with the AWS provider. If someone deletes the NodePool this triggers deletion of all NodeClaims but the NodePool is allowed to delete first. This can result in this hanging indefinitely.